.changeset/loud-items-joke.md

```diff
@@ -0,0 +1,34 @@
+---
+"fluid-framework": minor
+"@fluidframework/tree": minor
+---
+---
+"section": "tree"
+---
+`Tree.schema` now returns `TreeNodeSchema`.
+
+
+The typing of `Tree.schema` has changed from:
+
+```typescript
+schema<T extends TreeNode | TreeLeafValue>(node: T): TreeNodeSchema<string, NodeKind, unknown, T>;
+```
+
+to:
+
+
+```typescript
+schema(node: TreeNode | TreeLeafValue): TreeNodeSchema;
+```
+
+The runtime behavior is unaffected: any code which worked and still compile is fine and does not need changes.
+
+`Tree.schema` was changed to mitigate two different issues:
+
+1. It tried to give a more specific type based on the type of the passed in value. When the type of the input is not know precisely (for example its a union of node types like `Foo | Bar`, or `TreeNode` or even `TreeNode | TreeLeafValue`), this was fine since schema are covariant over their node type. However when the input was more specific that the schema type, for example the type is simply `0`, this would result in unsound typing, since the create function could actually return values that did not conform with that schema (for example `schema.create(1)` for a the number schema typed with `0` would return `1` with type `0`).
```

Comment:
```suggestion
1. It tried to give a more specific type based on the type of the passed in value. When the type of the input is not know precisely (for example if it's a union of node types like `Foo | Bar`, or `TreeNode` or even `TreeNode | TreeLeafValue`), this was fine since schema are covariant over their node type. However when the input was more specific than the schema type, for example if the type is simply `0`, this would result in unsound typing, since the create function could actually return values that did not conform with that schema (for example `schema.create(1)` for a number schema typed with `0` would return `1` with type `0`).
```